### DISCUSSION TITLE
how to set the input range in nntool?

### DISCUSSION THREAD

**Discussion by noahzn** [bitcraze/discussions#573] [Q&A] [answered]
[update 2] I checked the output[0][0] which should be the dumped input image, but it's not the correct image. I can only see strips [images](https://user-images.githubusercontent.com/8244850/220323985-53dcdc89-dc12-4958-8af0-de62a7ead575.png)

[update 1] I use the nntool to get a float model `(aquant --scheme FLOAT)`, the result is not correct, either.

Hi, I'm quantizing my custom model using nntool. My network is trained with PyTorch, and the input data is scaled to [0, 1] before training. I tested the dumped data using float output by nntool, but the result is not correct. I suspect that it's caused by the input range. 

I use `set_input_divisor 255`, and then `qshow`, the `out` of `input_1` is `-0.93<(i8-0.00)*0.00728120<0.92 forced`. How should I make it in the range of [0, 1]? 

Thanks!

**Comment by noahzn** [comment, marked as answer]
Using `set_input_divisor 255` can scale the input image into [0,1], and this can be verified if you dump a result into a .npy file. You can verify the min() and max() of the output[0][0].

I recompiled a newer nntool, and converted my one-channel gray-scale .jpg to three channels. Now it works.

**Reply by knmcguire** [reply]
thanks for providing the answer!
